napari/utils/transforms/transform_utils.py

```diff
@@ -490,3 +490,66 @@ def is_diagonal(matrix, tol=1e-8):
         return np.count_nonzero(non_diag) == 0
     else:
         return np.max(np.abs(non_diag)) <= tol
+
+
+def get_permutation(matrix, tol=1e-8):
+    """Determine the axis permutation order for an affine.
+
+    Parameters
+    ----------
+    matrix : 2-D array
+        The matrix to check.
+    tol : float, optional
+        Consider any entries with magnitude < tol as 0.
+
+    Returns
+    -------
+    c : tuple of int
+        A tuple of integers where ``c[i]`` is the index of the column with a
+        non-zero entry for row ``i``. If we were to set the non-zero entries
+        in `affine` to 1 and remove any translation, the operation of the affine
+        on a data matrix would be equivalent to ``data.transpose(c)``.
+    """
+    n = matrix.shape[0]
+    if tol == 0:
+        r, c = np.nonzero(matrix)
+    else:
+        r, c = np.nonzero(np.abs(matrix) >= tol)
+    # permutation matrix will only have 1 entry per column
+    c = tuple(c)
+    c_check = len(set(c)) == n
+    if not c_check:
```

Comment:
```suggestion
    if len(set(c)) != n:
```

Optional: simplify or at least consider a more meaningful name than `c_check`.